metricproviders/datadog/datadog_test.go

```diff
@@ -0,0 +1,189 @@
+package datadog
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRunSuite(t *testing.T) {
+
+	const expectedApiKey = "0123456789abcdef0123456789abcdef"
+	const expectedAppKey = "0123456789abcdef0123456789abcdef01234567"
+
+	unixNow = func() int64 { return 1599076435 }
+
+	// Test Cases
+	var tests = []struct {
+		webServerStatus      int
+		webServerResponse    string
+		metric               v1alpha1.Metric
+		expectedValue        string
+		expectedPhase        v1alpha1.AnalysisPhase
+		expectedErrorMessage string
+	}{
+		// When last value of time series matches condition then succeed.
+		{
+			webServerStatus:   200,
+			webServerResponse: `{"status":"ok","series":[{"pointlist":[[1598867910000,0.0020008318672513122],[1598867925000,0.0003332881882246533]]}]}`,
+			metric: v1alpha1.Metric{
+				Name:             "foo",
+				SuccessCondition: "asFloat(result) < 0.001",
+				FailureCondition: "asFloat(result) >= 0.001",
```

Comment:
Why is `asFloat` necessary when the return values already appear as floats?

Reply:
It's using jsonpath - same as the web metric to extract the result, so it comes out as a string. The metric provider could convert it back to a float automatically. I can't imagine they'd ever be a query that returns something that isn't a float (but as it is we avoid that assumption anyway).

Reply:
Hmmm, I didn't realize that about the jsonpath library. I think we better take a different approach than the webmetric, and avoid using `k8s.io/client-go/util/jsonpath` if we know that it has the limitation of always returning strings.

Flagger appears to make the assumption that the pointlist returns a float. I think we can do the same.
https://github.com/weaveworks/flagger/blob/ffbbc2ca3384509f71791bc3aa476e0cc51845eb/pkg/metrics/providers/datadog.go#L42-L45

Can you refactor such that, `asFloat()` will not be necessary?

BTW, I've filed https://github.com/argoproj/argo-rollouts/issues/728 so that web metric preserves the original format, and `asFloat()` would only be necessary if the return value was a string.